Code/FooBar.php

```diff
@@ -6,19 +6,27 @@ class FooBar
 {
     public function checker($number)
     {
-        if (is_int($number) && $number > 0) {
-            if ($number % 3 == 0 && $number % 5 == 0) {
-                $result = "Foo, Bar";
-            } elseif ($number % 5 == 0) {
-                $result = "Bar";
-            } elseif ($number % 3 == 0) {
-                $result = "Foo";
-            } else {
-                $result = strval($number);
-            }
-            return $result;
-        } else {
+        $foo = "Foo";
+        $bar = "Bar";
+        $comma = ", ";
+
+        if (!is_int($number) || $number < 0) {
             throw new InvalidArgumentException();
+        } else {
+            switch ($number) {
```

Comment:
No, `switch` is not "faster" than `if`. Imagine that "$number % 3" will take 10 seconds. Your implementation results for 20 seconds if input is in order of 3, but not 5.